src/generators/ast.ts

```diff
@@ -127,6 +130,31 @@ export function getPropertiesSchema(event: Schema): ObjectTypeSchema {
   };
 }
 
+// get traits from the event schema
+export function getTraitsSchema(event: Schema): ObjectTypeSchema {
+  let traits: ObjectTypeSchema | undefined = undefined;
+
+  if (event.type === Type.OBJECT) {
+    const traitsSchema = event.properties.find(
+      (schema: Schema): boolean => schema.name === 'traits',
+    );
+
+    if (traitsSchema && traitsSchema.type === Type.OBJECT) {
+      traits = traitsSchema;
+    }
+  }
+
+  return {
+    type: Type.OBJECT,
+    properties: [],
+    ...(traits || {}),
+    isRequired: traits ? !!traits.isRequired : false,
+    isNullable: false,
+    name: event.name,
```

Comment:
Why does traits schema have an event name? Is it always undefined?

Reply:
This is the function name for the generator. In Tracking Plan we have the event names as identify, page, group, screen for non - track events